src/typegate/src/runtimes/patterns/messenger/async_messenger.ts

```diff
@@ -131,6 +89,33 @@ export class AsyncMessenger<Broker, M, A> {
     this.#tasks.delete(id);
   }
 
+  private startTaskTimer(task: Message<M>) {
+    return setTimeout(() => {
+      if (this.#tasks.has(task.id)) {
+        if (
+          task.remainingPulseCount !== undefined &&
+          task.remainingPulseCount > 0
+        ) {
+          // check again next time if unterminated
```

Comment:
A bit of context: the old implementation used only one timer and timeouts were approximated by timing out at the step+1 tick, so only a single timer to **easier** to cleanup/setup/keep track of.

A bit curious though, did you have any clues with what went wrong old implementation? Like why would it even timeout even though we are pulsating at a constant rate? 🤔